### PR TITLE
Tests: Migrate v1 algod dependencies to v2 in cucumber tests

### PR DESCRIPTION
**Pull request by algochoi**
Migrates v1 algod usage in cucumber integration tests to use v2. Also deletes some unused steps relating to v1 APIs.

SDK Testing PR: https://github.com/algorand/algorand-sdk-testing/pull/249

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="master"
+SDK_TESTING_BRANCH="remove-v1-tests"
```
Comment (algochoi) [inline .test-env:3]:
```suggestion
SDK_TESTING_BRANCH="master"
```

File: test/steps_test.go [hunk 2]
```diff
@@ -1555,37 +1422,54 @@ func checkExpectedVsActualAssetParams() error {
...
+		if idx == assetID {
+			found = true
+			foundAsset = asset
+			break
+		}
+	}
+	if !found {
+		return modelsV2.AssetHolding{}, fmt.Errorf("asset ID %d was not found", assetID)
+	}
+	return foundAsset, nil
```
Comment (ahangsu) [inline test/steps_test.go:1461]: nit: simplification, 

```suggestion
	for _, asset := range assets {
		if asset.AssetId == assetID {
			return asset, nil
		}
	}
	return modelsV2.AssetHolding{}, fmt.Errorf("asset ID %d was not found", assetID)
```

**Review by ahangsu — APPROVED** [review]
The change seems straightforward, looks generally good to me

File: test/steps_test.go (hunk 2 above)
Comment (algochoi) [reply test/steps_test.go:1461]: Nice catch, thanks!

**Pull request merged**